### ISSUE TITLE
[Vineyard Operator] Realize all the operations related to the vineyard of k8s through the vineyardctl API

### ISSUE THREAD

**Issue by dashanji** [alibaba/GraphScope#2539]
To make better use of the capabilities(sidecar, scheduler, etc) of the [vineyard operator](https://v6d.io/notes/cloud-native/vineyard-operator.html#vineyard-operator), we want to manage all the vineyard-related components in the current graphscope through the vineyardctl API(the API component of vineyard operator). It mainly includes the following operations.

- [ ] Replace the current sidecar with the sidecar API of vineyardctl 
- [ ] Realize the scheduling of various engines (GAE, GIE, and GLE) through the scheduler API of vineyardctl.
- [ ] Provide the upper engine with the ability to backup and restore the vineyard cluster through the backup and recover API of vineyardctl.

**Comment by dashanji** [comment]
@sighingnow @siyuan0322. Do you have any comments here?

**Comment by siyuan0322** [comment]
> Replace the current sidecar with the sidecar API of vineyardctl

I have no problem with that

> Realize the scheduling of various engines (GAE, GIE, and GLE) through the scheduler API of vineyardctl.

I think you mean to use a different scheduler? Althrough we could only use default scheduler, it's trivial to make `schedulerName` configurable, which part could vineyardctl play in this context?

> Provide the upper engine with the ability to backup and restore the vineyard cluster through the backup and recover API of vineyardctl.

I thought GraphScope doesn't manage the vineyard cluster, it's ought to be managed by some cluster admin. Why make engines do such management 😂

**Comment by dashanji** [comment]
> I think you mean to use a different scheduler? Althrough we could only use default scheduler, it's trivial to make `schedulerName` configurable, which part could vineyardctl play in this context?

It is mainly used in a workflow(GAE->GIE->GLE). Actually, we don't need to setup the `schedulerName` for graphscope engines. As vineyardctl supports scheduling the engines outside the cluster which will find the most fittable vineyard node and setup the relevant `nodeName` for engines. For more detailers about the scheduler, please refer to [vineyard scheduler](https://v6d.io/notes/cloud-native/vineyard-operator.html#vineyard-scheduler).

**Comment by dashanji** [comment]
> I thought GraphScope doesn't manage the vineyard cluster, it's ought to be managed by some cluster admin. Why make engines do such management 😂

Yeah, It's optional, users can also manage the vineyard cluster outside GraphScope. The thought is only to provide some configurations for engines.

**Comment by siyuan0322** [comment]
> Yeah, It's optional, users can also manage the vineyard cluster outside GraphScope. The thought is only to provide some configurations for engines.

Could I see some examples?

**Comment by dashanji** [comment]
Yeah, It's managed by the CRD `Backup` and `Recover` referring the [doc](https://v6d.io/notes/cloud-native/vineyard-operator.html#failover-mechanism-of-vineyard-cluster). If a session of GraphScope wants to backup or recover a vineyard cluster, we could simply create or delete the relevant CR here.
